portfolio/src/main/java/com/google/sps/servlets/DisplayBlobServlet.java

```diff
@@ -0,0 +1,44 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This servlet is called to display images stored in Blobstore given their blobKey
+ */
+@WebServlet("/display-blobstore")
+public class DisplayBlobServlet extends HttpServlet {
```

Comment:
So this servlet _retrieves_ the data from BlobStore right?  It is the browser that does the actual displaying (and even then, it might not be actually displaying the image - it might be downloading/saving it for example).  Its always a good idea to keep "separation of concerns" at the front of your mind: this servlet only exists to *get the data* from blobstore and nothing else... the other parts of the system will decide if it is displayed or saved or processed some other way etc.

So it might make sense to call it something like `RetrieveBlobServlet`?  Up to you.

Reply:
Thank you for the notice, I changed the name to `RetrieveBlobServlet.`